dracut/03flatcar-network/afterburn-network-kargs.service

```
@@ -0,0 +1,24 @@
+[Unit]
+Description=Afterburn Initrd Setup Network Kernel Arguments
+Documentation=https://coreos.github.io/afterburn/usage/initrd-network-cmdline/
+
+# This service may produce additional kargs fragments,
+# which are then consumed by dracut-cmdline(8).
+DefaultDependencies=no
+Before=dracut-cmdline.service systemd-networkd.service
+PartOf=systemd-networkd.service
+# For extra safety
+ConditionKernelCommandLine=|coreos.oem.id=vmware
+ConditionKernelCommandLine=|flatcar.oem.id=vmware
+OnFailure=emergency.target
+OnFailureJobMode=isolate
+
+[Service]
+Environment=AFTERBURN_OPT_PROVIDER=--cmdline
+Environment=AFTERBURN_NETWORK_KARGS_DEFAULT=''
+ExecStart=/usr/bin/coreos-metadata exp rd-network-kargs ${AFTERBURN_OPT_PROVIDER} --default-value ${AFTERBURN_NETWORK_KARGS_DEFAULT}
```

Could you add a comment about the reason to have AFTERBURN_OPT_PROVIDER and AFTERBURN_NETWORK_KARGS_DEFAULT instead of having those values inline in ExecStart? Is this for drop-in files?

---

I basically copied this service file from coreos/afterburn repository: https://github.com/coreos/afterburn/blob/main/dracut/30afterburn/afterburn-network-kargs.service and followed their way to do. 
For example, the `AFTERBURN_NETWORK_KARGS_DEFAULT` is defined later in: https://github.com/coreos/fedora-coreos-config/blob/testing-devel/overlay.d/05core/usr/lib/dracut/modules.d/35coreos-network/50-afterburn-network-kargs-default.conf#L7 and this value could be platform specific.

---

Yeah, so my understanding is that upstream did not define `AFTERBURN_NETWORK_KARGS_DEFAULT` in their service file, because downstream should define it through a drop-in file, just like it's done in the links you posted. Also, we are not using the service file from upstream, instead we have our own, tailored for our needs. So I'd just change it to something like:

```suggestion
ExecStart=/usr/bin/coreos-metadata exp rd-network-kargs --cmdline --default-value ''
```

This is basically the same thing. From my cursory search, neither [`AFTERBURN_NETWORK_KARGS_DEFAULT`](https://github.com/flatcar-linux/afterburn/search?q=AFTERBURN_NETWORK_KARGS_DEFAULT) nor [`AFTERBURN_OPT_PROVIDER`](https://github.com/flatcar-linux/afterburn/search?q=AFTERBURN_OPT_PROVIDER) has no special meaning within afterburn.

But if it is done with the aim of deviating from upstream in the most minimal way, then I have no problem with it staying as is.

---

I'll go ahead and commit your suggestion - even if one wants to add a dropin to update `AFTERBURN_NETWORK_KARGS_DEFAULT` for example it will be useless since this service runs in the initramfs only.